anaDACScan.py

```diff
@@ -0,0 +1,236 @@
+             dict_dacScanResults[oh][vfat].GetXaxis().SetTitle(nameY)
+             dict_dacScanResults[oh][vfat].GetYaxis().SetTitle(nameX)
+
+    outputFiles = {}         
+             
+    for oh in ohArray:         
+        outputFiles[oh] = r.TFile(dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate+"/"+args.outfilename,'recreate')
+             
+    # Loop over events in the tree and fill plots
+    for event in dacScanFile.dacScanTree:
+        oh = event.link
+        vfat = event.vfatN
+
+        #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+        calibrated_DAC_value=calInfo[oh]['slope'][vfat]*event.dacValX+calInfo[oh]['intercept'][vfat]
+        calibrated_DAC_error=calInfo[oh]['slope'][vfat]*event.dacValX_Err+calInfo[oh]['intercept'][vfat]
+
+        if args.assignXErrors:
+            dict_dacScanResults[oh][vfat].SetPoint(dict_dacScanResults[oh][vfat].GetN(),event.dacValY,calibrated_DAC_value)
+            dict_dacScanResults[oh][vfat].SetPointError(dict_dacScanResults[oh][vfat].GetN()-1,event.dacValY_Err,calibrated_DAC_error)
+        else:    
+            dict_dacScanResults[oh][vfat].SetPoint(dict_dacScanResults[oh][vfat].GetN(),event.dacValY,calibrated_DAC_value)
+            dict_dacScanResults[oh][vfat].SetPointError(dict_dacScanResults[oh][vfat].GetN()-1,event.dacValY_Err,0)
+
+    # Fit the TGraphErrors objects    
+    for oh in ohArray:
+        for vfat in range(0,24):
+            dict_dacScanFuncs[oh][vfat] = r.TF1("DAC Scan Function","[0]*x+[1]")
+            dict_dacScanResults[oh][vfat].Fit(dict_dacScanFuncs[oh][vfat]) 
+
+    nominal = -1        
+
+    #from Tables 9 and 10 of the VFAT3 manual 
+    if nameX == "CFG_THR_ARM_DAC":
+        nominal = 3.2
+    else:
+        print("Error: unexpected value of nameX: '%s'"%nameX)
+        exit(1)
+        
+    # Determine DAC values to achieve recommended bias voltage and current settings
+    graph_dacVals = {}
+    tree_dacVals = {}
+    dict_dacVals = nesteddict()
+    for oh in ohArray:
+        graph_dacVals[oh] = r.TGraph()
+        tree_dacVals[oh] = r.TTree()
+        vfatN_branch = np.zeros(1, dtype=int)
+        dacVal_branch = np.zeros(1, dtype=int)
+        tree_dacVals[oh].Branch('vfatN',vfatN_branch,'vfatN/i')
+        tree_dacVals[oh].Branch('dacVal',dacVal_branch,'dacVal/i')
+        for vfat in range(0,24):
+             dict_dacVals[oh][vfat] = dict_dacScanResults[oh][vfat].Eval(nominal)
+             graph_dacVals[oh].SetPoint(graph_dacVals[oh].GetN(),vfat,dict_dacVals[oh][vfat])
+             vfatN_branch[0] = vfat
+             dacVal_branch[0] = dict_dacVals[oh][vfat]
+             tree_dacVals[oh].Fill() 
+             
+    # Write out the dacVal results to both a file and the terminal         
+    print( "| OH | vfatN | dacVal |")
+    print( "| -- | ----- | ------ |")        
```

Comment:
To make reading this easier if pasted into mattermost change line 218 to:

```python
print("| :-: | :---: | :----: |")
```

This will ensure the columns are centered, see for example [markdown table documentation](https://docs.mattermost.com/help/messaging/formatting-text.html#tables).

Reply:
done